zucchini-ui-backend/src/main/java/io/zucchiniui/backend/scenario/views/ScenarioViewAccess.java

```diff
@@ -190,4 +191,38 @@ public ScenarioStats getStats(final Consumer<ScenarioQuery> preparator) {
             .collect(Collectors.toSet());
     }
 
+    public List<GroupedStepsListItemView> getStepDefinitions(final Consumer<ScenarioQuery> preparator) {
+
+        final Query<Scenario> query = scenarioDAO.prepareTypedQuery(preparator).project("steps", true);
+        List<Step> steps = MorphiaUtils.streamQuery(query)
+            .flatMap(scenario -> scenario.getSteps().stream())
+            .collect(Collectors.toList());
+
+        List<GroupedStepsListItemView> groupedSteps = new ArrayList<>();
+        steps.stream().filter(step -> step.getStepDefinitionLocation() != null).
```

Comment:
Last dot with the `.forEach`

Reply:
You can apply the filter between the flatMap and collect operations of the `MorphiaUtils.streamQuery(query)` stream.